hivemind/utils/networking.py

```diff
@@ -36,3 +40,17 @@ def find_open_port(params=(socket.AF_INET, socket.SOCK_STREAM), opt=(socket.SOL_
             return sock.getsockname()[1]
     except Exception as e:
         raise e
+
+
+def choose_ip_address(maddrs: Sequence[Multiaddr],
+                      prefer_global: bool = True,
+                      protocol_priority: Sequence[str] = ('ip4', 'ip6')) -> Hostname:
+    for need_global in [prefer_global, not prefer_global]:
+        for protocol in protocol_priority:
+            for addr in maddrs:
+                if protocol in addr.protocols():
+                    value = addr[protocol]
```

Comment:
I'll rename it to `value_for_protocol` since `addr[protocol]` is an alias for `addr.value_for_protocol(protocol)` and since other names (`addr`, `ip_address`) are already used.